runtime/compiler/local_activations.go

```diff
@@ -198,7 +214,7 @@ func (a *LocalActivations) Pop() {
 func (a *LocalActivations) CurrentOrNew() *LocalActivation {
 	current := a.Current()
 	if current == nil {
-		return NewLocalActivation(nil)
+		return NewLocalActivation(a.memoryGauge, nil)
```

Comment:
Unrelated to this particular PR:
It's interesting that the tests do not exercise this code path and similar cases below. 

I guess because we always have an initial activation in the activations before using/accessing the activations? 
Maybe we can remove the nil cases.

Reply:
Yup, might be possible. I'll add this in a separate PR with some tests, to keep this one to a minimal changes